sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -538,7 +539,10 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    * @since 2.0.0
    */
   @scala.annotation.varargs
-  def csv(paths: String*): DataFrame = format("csv").load(paths : _*)
+  def csv(paths: String*): DataFrame = {
+    require(paths.nonEmpty, "The paths cannot be empty")
```

Comment:
While I agree with the intention of setting the behaviour down, I concern about behaviour changes. Empty Dataset is a concept existent, e.g., `SparkSession.emptyDataset`.